modules/apps/frontend-data-set/frontend-data-set-web/src/main/resources/META-INF/resources/utils/index.js

```diff
@@ -266,7 +266,14 @@ export function loadData(
 		);
 	}
 
-	return executeAsyncAction(url, 'GET').then((response) => response.json());
+	const response = await executeAsyncAction(url, 'GET');
+	const responseJSON = await response.json();
+
+	return {
+		data: responseJSON,
+		ok: response.ok,
+		status: response.status,
+	};
```

Comment:
Nice use of `async` / `await`! :sparkles:  I also like how we are exposing both data and response attributes. I never liked this in vanilla JS `fetch` API, that we need to split response-handling logic into two `then` blocks.

Sidenote - we probably want to get rid of `executeAsyncAction`, replacing it with inline `fetch`. It seems to me like a bit of an unnecessary overengineering. What do you think? This can be a separate task.

Reply:
Thxs!
Agree about the two mandatory promises to get a _proper_ response

And totally agree about omit/delete the _executeAsyncAction_ function, I can not find any advantages here!
I'll create a task for it